bokeh/application/handlers/directory.py

```diff
@@ -14,12 +14,14 @@ def __init__(self, *args, **kwargs):
             raise ValueError('Must pass a filename to DirectoryHandler')
 
         src_path = kwargs['filename']
-        mainpy = os.path.join(src_path, 'main.py')
-        if not os.path.exists(mainpy):
-            raise ValueError("No 'main.py' in %s" % (src_path))
+        main = os.path.join(src_path, 'main.py')
+        if not os.path.exists(main):
+            main = os.path.join(src_path, 'main.ipynb')
+            if not os.path.exists(main):
+                raise ValueError("No 'main.py' or 'main.ipynb' in %s" % (src_path))
```

Comment:
worth complaining if the app has both main.py and main.ipynb? "main.ipynb ignored" kind of message